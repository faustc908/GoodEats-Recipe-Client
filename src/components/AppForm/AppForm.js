import React from "react";

// Present search form

const AppForm = (props) => {
  const inputEl = React.useRef(null);

  // Get value of entry and pass to onSearch

  const handleSearch = (e) => {
    e.preventDefault();
    props.onSearch(inputEl.current.value);
  };

  return (
    <form onSubmit={handleSearch} className="search-form">
      <input
        className="search-bar"
        type="text"
        placeholder="Banana"
        ref={inputEl}
        required
      />
      <button className="search-button" type="submit">
        Search
      </button>
    </form>
  );
};

export default AppForm;
