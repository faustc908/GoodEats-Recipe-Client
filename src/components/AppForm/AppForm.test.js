import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AppForm from './AppForm'
import ReactDOM from 'react-dom';

describe(`AppForm component`, () => {

    it('renders a AppForm by default', () => {
        const wrapper = shallow(<AppForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AppForm />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
})