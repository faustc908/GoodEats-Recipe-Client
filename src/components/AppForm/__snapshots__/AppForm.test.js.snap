// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppForm component renders a AppForm by default 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <input
    className="search-bar"
    placeholder="Banana"
    required={true}
    type="text"
  />
  <button
    className="search-button"
    type="submit"
  >
    Search
  </button>
</form>
`;
